detection_rules/version_lock.py

```diff
@@ -106,13 +106,13 @@ def add_changes(r, *msg):
                 min_stack = _convert_lock_version(rule.contents.metadata.min_stack_version)
 
                 current_rule_lock = rule.contents.lock_info(bump=not exclude_version_update)
-                existing_rule_lock: dict = current_version_lock.setdefault(rule.id, {})
+                existing_rule_lock: dict = lock_file_contents.setdefault(rule.id, {})
                 original_hash = existing_rule_lock.get('sha256')
 
                 # prevent rule type changes for already locked and released rules (#1854)
-                if current_rule_lock:
+                if existing_rule_lock:
                     name = current_rule_lock['rule_name']
-                    existing_type = current_rule_lock['type']
+                    existing_type = existing_rule_lock['type']
```

Comment:
This was referencing the _new_ lock info rather than the _existing_ (lock file) lock info as expected